examples/solitaire/src/SolitairePuzzle.tsx

```diff
@@ -106,29 +109,41 @@ export const Puzzle = ({
         cards.push(column);
         currentIndex += i;
       }
+
+      const newState: HistoryState = {
+        flippedCards: [],
+        stockCards: shuffledCards.slice(28), 
+        suitCards: state.suitCards,
+        tableauCards: cards, 
+        wasteCards: state.wasteCards,
+      };
+      setGameHistory((prev) => {
+        const state = [...(prev ?? []), newState];
+        onProgress?.(JSON.stringify(state));
+        return state;
+      });
+
       return {
         ...prev,
         stockCards: shuffledCards.slice(28),
         tableauCards: cards,
       };
-    });
+    }); 
+  // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
you remove this after you add `state` as a dependency

```suggestion
```

Reply:
Even when I add the state, it still throws a warning about a missing dependency: "onProgress." But if I add onProgress, things get weird.